lib/dialects/sqlite/query-generator.js

```diff
@@ -79,8 +82,139 @@ const QueryGenerator = {
     return !!value ? 1 : 0;
   },
 
-  addColumnQuery(table, key, dataType) {
+  handleSequelizeMethod(smth, tableName, factory, options, prepend) {
+    if (smth instanceof Utils.Json) {
+      // Parse nested object
+      if (smth.conditions) {
+        const conditions = _.map(this.parseConditionObject(smth.conditions), condition =>
+          `json_extract(${this.quoteIdentifier(_.first(condition.path))}, '$.${_.tail(condition.path).join('.')}') = '${condition.value}'`
+        );
+
+        return conditions.join(' and ');
+      } else if (smth.path) {
+        let str;
+
+        // Allow specifying conditions using the sqlite json1 functions
+        // https://sqlite.org/json1.html
+        if (/(json(?:_.*)*)\(.*\)/i.test(smth.path)) {
```

Comment:
can you have nested parenthesis in the input, e.g using the result of a function as a parameter to the JSON function?
In that case a regexp will not suffice.
Shouldn't you pass a function call as `sequelize.fn()`?
Can this cause SQL injection if `where` comes from user input?